go/diff/apply_patch.go

```diff
@@ -0,0 +1,287 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package diff
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/attic-labs/noms/go/d"
+	"github.com/attic-labs/noms/go/types"
+)
+
+// ApplyPatch applies a Patch (list of diffs) to a graph. It makes the following
+// contract:
+//  Given 2 Noms graphs: a1 and a2:
+//    ApplyPatch(a1, Diff(a1, a2)) == a2
```

Comment:
Diff() doesn't actually return anything, but if it did, i would return []Difference which I've typedef'd to path: 'type Patch []Difference'.
I think ApplyPatch() gives some useful context and Apply() is too general. Let me know if you still want this changed.